src/SuperSocket.Kestrel/KestrelPipeConnection.cs

```diff
@@ -94,4 +89,11 @@ private void OnConnectionClosed()
     {
         Cancel();
     }
+
+    protected override Task StartInputPipeTask<TPackageInfo>(IObjectPipe<TPackageInfo> packagePipe,
+        CancellationToken cancellationToken)
+    {
+        cancellationToken.Register(SupplyEnd);

```

Comment:
packagePipe can be read here, so yo don't need to define a field for packagePipe in the base class.